backend/remote-state/artifactory/client.go

```diff
@@ -48,16 +79,88 @@ func (c *ArtifactoryClient) Get() (*remote.Payload, error) {
 }
 
 func (c *ArtifactoryClient) Put(data []byte) error {
+	var err error
+	logStart("Put")
+	defer logResult("Put", &err)
+
 	p := fmt.Sprintf("%s/%s/%s", c.repo, c.subpath, ARTIF_TFSTATE_NAME)
-	if _, err := c.nativeClient.Put(p, string(data), make(map[string]string)); err == nil {
+	if _, err = c.nativeClient.Put(p, string(data), make(map[string]string)); err == nil {
 		return nil
 	} else {
 		return fmt.Errorf("Failed to upload state: %v", err)
 	}
 }
 
 func (c *ArtifactoryClient) Delete() error {
+	var err error
+	logStart("Delete")
+	defer logResult("Delete", &err)
 	p := fmt.Sprintf("%s/%s/%s", c.repo, c.subpath, ARTIF_TFSTATE_NAME)
-	err := c.nativeClient.Delete(p)
+	err = c.nativeClient.Delete(p)
+	return err
+}
+
+func (c *ArtifactoryClient) Lock(info *state.LockInfo) (string, error) {
+	if c.lockUrl == "" {
+		return "", nil
+	}
+	c.lockID = ""
+	var err error
+	var output []byte
+	logStart("Lock")
+	defer logResult("Lock", &err)
+	rand.Seed(time.Now().UnixNano())
```

Comment:
A PRNG should only be seeded once, this makes the output more predictable. The provided `info` should already contain a cryptographically secure ID.